ideal_device/io/abstract_io.h

```diff
@@ -12,16 +12,17 @@ namespace tt::umd {
 // This abstracts all readers and writers.
 // A user can hold this object to bypass the need to go through regular interfaces and 
 class AbstractIO {
-    public:
-    void write_u8(uint32_t address, uint8_t value);
-    void write_u32(uint32_t address, uint32_t value);
-    void write(uint32_t address, uint8_t* arr_ptr, uint32_t size);
-    void write(uint32_t address, uint32_t* arr_ptr, uint32_t size);
 
-    uint8_t read_u8(uint32_t address);
-    uint32_t read_u32(uint32_t address);
-    void read(uint32_t address, uint8_t* arr_ptr, uint32_t size);
-    void read(uint32_t address, uint32_t* arr_ptr, uint32_t size);
+public:
```

Comment:
I suggest making addresses `uint64_t` and sizes `size_t`.  There are address spaces in Blackhole where > 32 bits are necessary.

```
    virtual void write(uint32_t address, uint8_t* arr_ptr, uint32_t size) = 0;
    virtual void write(uint32_t address, uint32_t* arr_ptr, uint32_t size) = 0;
```
I think this would be less ambiguous as a single method that takes a `const void*`.  From those two signatures, it is not clear if `size` means "number of elements" or "bytes".

Reply:
Makes sense, void* is less ambiguous.